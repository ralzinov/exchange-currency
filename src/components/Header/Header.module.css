.host {
    display: flex;
    background: #24292e;
    padding: 12px 0;
    color: #fff;
    min-height: 40px;
    align-items: center;
}
