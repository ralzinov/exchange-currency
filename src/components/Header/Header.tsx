import * as React from 'react';
import {ReactNode} from 'react';
import styles from './Header.module.css';

interface IHeaderProps {
    children: ReactNode|ReactNode[];
}

export const Header = (props: IHeaderProps) => {
    return (
        <header className={styles.host}>{props.children}</header>
    );
};
