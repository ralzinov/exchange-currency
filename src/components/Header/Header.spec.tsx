import React from 'react';
import ReactDOM from 'react-dom';
import {Header} from './Header';
import {create} from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header>abc</Header>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should render with children', () => {
    const button = create(
        <Header>Label</Header>
    );
    expect(button.toJSON()).toMatchSnapshot();
});
