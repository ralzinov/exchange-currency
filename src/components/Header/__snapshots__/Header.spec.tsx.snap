// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with children 1`] = `
<header
  className="host"
>
  Label
</header>
`;
